pkg/secret/secret.go

```diff
@@ -0,0 +1,186 @@
+package secret
+
+import (
+	"fmt"
+	"github.com/appuio/seiso/pkg/kubernetes"
+	"github.com/appuio/seiso/pkg/openshift"
+	"github.com/appuio/seiso/pkg/util"
+	log "github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sort"
+	"time"
+)
+
+type (
+	Service interface {
+		ListNamesAndLabels() (resourceNames, labels []string, err error)
```

Comment:
It seems `ListNamesAndLabels` is only ever used just for logging when the user did not specify a label selector. I think it makes more sense to move the printing logic to the service with a new name `PrintNamesAndLabels` and use a log line per each entry. Right now the entries are printed as an array, where no association between which label goes with which resource. Example:
```
INFO Please use labels to select Secrets. The following Secrets and Labels are available:  
 - namespace=redacted
 - 🎫 labels="[certmanager.k8s.io/certificate-name=redacted]" 
 - 🔐 secrets="[builder-dockercfg-5bbzr builder-token-s8d69 builder-token-x6b4g cloudflare-api-key default-dockercfg-q2t69 default-token-j8c69 default-token-jmqpv deployer-dockercfg-frrlq deployer-token-52ncz deployer-token-lm6sl letsencrypt-prod operator-auth]"
```
First of all, it seems the label list is way too short, there are way more. Secondly, I cannot make out which secret has which labels.

How about printing each secret with their label per log line instead of 1 log statement? This way it's clear which secret has which labels.

Something along the lines
```go
log.Info("Following Secrets are available in namespace <namespace>")
for each secret {
  log.Info("Name: <secret.Name>, labels: [<label1=key1>,<label2=key2>]")
}
```

Reply:
I don't like the fact that I'm changing everything in secrets and configmaps in this MR, it was suppose to add unit tests.... For example the previous issue you raised is directly linked to this function. I added a new test there and fixed the error, here you asked me to change completely the signature. After I change the signature I have to go back to the tests to change what I already fixed in the previous issue. Can we just limit how much refactoring I'm doing? otherwise I won't close this merge request. We can open small tasks to improve it.

Reply:
We agreed to update an inner function, add unit tests to it and move the logs inside this function.